NewForce-Cohort-7/tabloidfullstack-numpty-buggers
# Sl view current users posts

Description
This pull request is for TabloidFullStack Trello ticket number: 4 , https://trello.com/c/hQDbUodm/4-4-view-current-users-posts, View Current User's Posts.
A user is now able to click on "My Posts" in that navbar and be directed to a page, seeing a list of all their own posts. Users can see the title, the name of the author, and the category.

The branch for the pull request: sl-viewCurrentUsersPosts

Updates

- Updated Post.cs, PostRepository.cs, IPostRepository.cs, PostController.cs (C# Files)
- Updated Post.js, PostManager.js (React.js Files)
- Added UserPosts.js
- Updated Header.js to have the nav item /myPosts
- Updated ApplicationViews.js to contain the /myPosts
- Also updated it so that the header for posts and my posts does not repeat itself

Type of change

New Feature - non-breaking

Testing Instructions
Follow these instructions to protect the branch you are working on and to test this pull request: https://gist.github.com/jordan-castelloe/96f76d5bd4cfbd2181084af674c2a1af#when-you-need-to-test-a-teammates-pr

Seed The Data
If you do not currently have the code for this project, clone the repository - (https://github.com/NewForce-Cohort-7/tabloidfullstack-numpty-buggers)

Open the TabloidFullStack.sln in visual studio and prepare your database.
In the github repository, open the SQL folder: https://github.com/NewForce-Cohort-7/tabloidfullstack-numpty-buggers/tree/main/SQL)
there are two files - [01_Tabloid_Create_DB.sql](https://github.com/NewForce-Cohort-7/tabloidfullstack-numpty-buggers/blob/main/SQL/01_Tabloid_Create_DB.sql), [02_Tabloid_Seed_Data.sql](https://github.com/NewForce-Cohort-7/tabloidfullstack-numpty-buggers/blob/main/SQL/02_Tabloid_Seed_Data.sql)

In Visual Studio in the top navbar click View and select SQL Server Object Explorer. Open the server marked as "SQLEXPRESS" in the SQL Server Object Explorer on the left side of Visual Code. Once there, add a new SQL server by selecting the 'Add SQL Server' option at the top of the side bar. Copy the data from the seed data and execute by pressing the green Play button.
start the server by clicking the solid green play button in the middle top of your screen. A browser window will display with the Tabloid landing page.

Test the View Current User Posts:
Once in the branch sl-viewCurrentUsersPosts and having booted up VS and performed "npm start" at the tabloid folder you should have swagger and localhost:3000 pulled up, login as [foo@bar.com] (no password needed)
click on "My Posts" in the navbar and you should see a list of all the posts

SQL Query test
If you would like to further test this or test it through a SQL query, create a query for the database you created at the start of the pull request testing. In your new query, paste:

SELECT p.Id AS PId, p.Title, p.Content, p.ImageLocation AS PImage, p.CreateDateTime AS PCreateDate, p.PublishDateTime, p.IsApproved, p.CategoryId, p.UserProfileId, up.Id AS UId, up.DisplayName, up.FirstName, up.LastName, up.Email, up.CreateDateTime AS UPCreateDate, up.ImageLocation AS UPImage, up.UserTypeId, c.Id AS CId, c.[Name] FROM Post p                        
LEFT JOIN UserProfile up ON up.Id = p.UserProfileId  
LEFT JOIN Category c ON p.CategoryId = c.Id
Where p.UserProfileId = 1
ORDER BY p.CreateDateTime

Run the test (green play button to the left of the query) You will see all posts made by user 1 "Foo Barrington" with all the other columns that come with it in the database
